src/model/document.ts

```diff
@@ -268,14 +280,21 @@ export class Document implements Feature {
   }
 
   /**
-   * Adds and indexes a feature to this documentled before resolve().
+   * Return all warnings on local features contained within the document as well
+   * as warnings on features detected by following through all imports. If the
+   * deep argument is set to false, only return warnings found on local
+   * features.
    */
-  _addFeature(feature: Feature) {
-    if (this._doneResolving) {
-      throw new Error('_addFeature can not be called after _resolve()');
+  getWarnings(deep?: boolean): Warning[] {
+    const warnings: Warning[] = [];
+    if (deep == null) {
+      deep = true;
```

Comment:
arg, I have pending comments I forgot to send, here where I'm favoring non-deep by default. I think this can change with other changes upcoming wrt resolving and querying. My logic here is that you don't expect to get warnings from a dependency when you ask for warnings from a file.

Outside of warnings I think we'll shortly need to refine what the concept of "deep" even is. So far we've been dealing with globally scoped features of documents, either a `customElements.define()` call, or a global symbol. We're going to have to adapt to scoped exports/imports, in which case, what does querying a document mean? Are you getting the exported symbols from a document, or the import symbols visible within a document, or all exported symbols reachable from a node in the dependency graph?

I'd argue that the last query isn't particularly useful most of the time (though it could be for some editor service suggestions), and so what you want by default is a shallow query for what's either exported from or imported into a document. Only getting local warnings would be consistent with this. Globals are kind of like having an `export * from x` for all x that a document imports, and so deep == false would be sort of like exports except re-exports. It could be that deep is to vague of a word for all these concepts.